pkgs/development/libraries/boost/cross-compile-170.patch

```diff
@@ -0,0 +1,12 @@
+--- a/bootstrap.sh
++++ b/bootstrap.sh
+@@ -223,7 +223,7 @@ rm -f config.log
+ if test "x$BJAM" = x; then
+-  $ECHO -n "Building Boost.Build engine with toolset $TOOLSET... "
++  $ECHO -n "Building Boost.Build engine... "
+   pwd=`pwd`
+-  (cd "$my_dir/tools/build/src/engine" && ./build.sh "$TOOLSET") > bootstrap.log 2>&1
++  (cd "$my_dir/tools/build/src/engine" && BUILD_CC="$CC_FOR_BUILD" ./build.sh cross-cc) > bootstrap.log 2>&1
```

Comment:
I don't know the whole context of the file here, but it is not possible to set BUILD_CC without patching?

Reply:
Seems like the tricky part is that boost-build itself does not support a "cross-cxx" toolset[1], so we somehow need to pass that only to the tools/build/src/engine/build.sh and not to ./bootstrap.sh.

At this point my only idea is that we could just do
```diff
diff --git a/pkgs/development/libraries/boost/generic.nix b/pkgs/development/libraries/boost/generic.nix
index abff1268bfa..d98812b47ba 100644
--- a/pkgs/development/libraries/boost/generic.nix
+++ b/pkgs/development/libraries/boost/generic.nix
@@ -149,6 +149,7 @@ stdenv.mkDerivation {
     cat << EOF >> user-config.jam
     using gcc : cross : ${stdenv.cc.targetPrefix}c++ ;
     EOF
+    sed '2iexport CXX=$CXX_FOR_BUILD' -i ./tools/build/src/engine/build.sh
   '';
 
   NIX_CFLAGS_LINK = lib.optionalString stdenv.isDarwin
```
or have `--with-bjam=${boost-build}/bin/b2`

[1] https://github.com/boostorg/build/tree/495b7ee9970787319dad6c70707ee8f4f18363e8/src/tools